astropy/utils/data.py

```diff
@@ -967,7 +965,7 @@ def compute_hash(localfn):
 
     Parameters
     ----------
-    localfn : str
+    localfn : str | Path
```

Comment:
This looks like a new feature? I think this one actually needs a change log.

Reply:
good point. I need to review this thoroughly myself before I undraft, and I'll keep this in mind when I do !